rfc/RullendeOppgraderingsvindu.md

```diff
@@ -0,0 +1,57 @@
+## Rullende oppgraderingsvindu på JustisHub
+
+**To be Reviewed By**: Martin Myran, Jostein Klakegg
+
+**Authors**: Carl Fredrik Bøkestad
+
+**Status**: Discussion
+
+### Problemstilling
+
+I forbindelse med endring av meldingsformater over JustisHub trenger vi rutine/system som gir forutsigbarhet for hvordan vi endrer/oppgraderer kontrakter.  Og som beskriver hvordan disse kan gjøres uten nedetid, i vanlige oppgraderings situasjoner, og når det implementerers _breaking changes_.
+
+Det er ulike virksomheter koblet til JustisHub det er derfor viktig at oppgraderinger kan gjøres autonomt og forutsigbart i størst mulig grad. Dette forutsetter at alle parter er inneforstått med regimet og de forpliktelser som følger av den beskrevne løsingen.
+
+
+### Løsningsforslag
+
+Forslaget går ut på at ved en oppgradering av kontrakt er mottaker/konsument ansvarlig for håndtere mottak av alle versjoner av meldingen i en _grace periode_.
+*Grace perioden* for et gitt miljø, blir man enige om på forhånd - f.eks. 1 måned (eller til at avsender har informert om at de ikke lenger kommer til å produsere tidligere versjoner).
+
+Header på meldingene må inneholde et versjonsnummer som tilsvarer versjonen på kontrakten meldingen skal reflektere slik at mottaker kan behandle meldingen korrekt (riktig versjon). Avsender/produsent av melding kan selv velge (innenfor grace-perioden) når man oppgraderer fra en versjon til en annen.
+
+Ved introduksjon  av ny versjon er det da mottaker sitt ansvar å kommunisere *End of life" på forrige versjon til avsender.
+
```

Comment:
Skulle vi også fått introdusert en måte å få oversikt i hvilke perioder kontrakter er gyldige? 
Eks en tabell som viser gyldigheten for en versjon, eks
| Versjon | Gyldig fra | Gyldig til |
|---------|------------|------------|
| 1.0     | 01.01.2023 | 15.06.2023 | 
| 1.1     | 16.06.2023 |            | 

Som kunne ha ligget i README for den enkelte kontrakt

Reply:
Jeg har lagt tabell i changelog.md